persistence-api/src/main/java/io/stargate/db/query/Condition.java

```diff
@@ -0,0 +1,130 @@
+package io.stargate.db.query;
+
+import static java.lang.String.format;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Preconditions;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.Column.ColumnType;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.immutables.value.Value.Immutable;
+
+@Immutable
+public interface Condition {
+  LHS lhs();
+
+  Predicate predicate();
+
+  @Nullable
+  TypedValue value();
+
+  /**
+   * Represents the left hand side of a condition.
+   *
+   * <p>This is usually a column, but technically can be:
+   *
+   * <ul>
+   *   <li>a column ("c = ...")
+   *   <li>a specific element in a map column ("c[v] = ...")
+   *   <li>a tuple of column name ("(c, d, e) = ...")
+   *   <li>the token of a tuple of column name ("TOKEN(c, d, e) = ...")
+   * </ul>
+   */
+  abstract class LHS {
+    enum Kind {
+      COLUMN,
+      MAP_ACCESS,
+      TUPLE,
+      TOKEN
+    }
+
+    private final Kind kind;
+
+    private LHS(Kind kind) {
+      this.kind = kind;
+    }
+
+    public Kind kind() {
+      return kind;
+    }
+
+    public abstract ColumnType valueType();
+
+    public boolean isUnset() {
+      return false;
+    }
+
+    public static LHS column(Column column) {
+      return new SimpleColumn(column);
+    }
+
+    public static LHS mapAccess(Column column, TypedValue key) {
+      return new MapAccess(column, key);
+    }
+
+    public static LHS columnTuple(List<Column> columns) {
+      // Not yet needed, but we should add it someday
+      throw new UnsupportedOperationException();
```

Comment:
As said above, I'm not a fan of polluting issue trackers. It's not really a task that adds value in itself. There is not strong reason to implement it until we need it, and when some code need it, it can just add it, not need for an issue.

But, that's just me. If you feel strongly an issue helps here, I'll create one.

Reply:
I think that having additional issues is not a problem and the team not forget that. So I am +1 on creating follow-up tickets